firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift

```diff
@@ -3319,18 +3319,43 @@ class BrowserViewController: UIViewController,
         if let url {
             switchToTabForURLOrOpen(url, isPrivate: isPrivate)
         } else {
-            guard let selectedTab = tabManager.selectedTab else { return }
-            if selectedTab.isPrivate == isPrivate, selectedTab.isFxHomeTab {
-                focusLocationTextField(forTab: selectedTab)
-            } else {
+            let isFocusLocationTextFieldOption = options?.contains(.focusLocationField) == true
+
+            if let selectedTab = tabManager.selectedTab {
```

Comment:
I think I would prefer having a similar solution [to this use case? ](https://github.com/mozilla-mobile/firefox-ios/blob/62780780e00d8fb28a2bcec6cfdc7105d6a05c85/firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift#L3359-L3371). Using a `guard` would enable us to keep the current function simple and easier to understand. I would also add a comment to explain why we need to wait for tab restoration